sycl/doc/extensions/experimental/sycl_ext_oneapi_complex.asciidoc

```diff
@@ -101,127 +111,236 @@ the decimal and an imaginary component equal to 0.
+  friend marray<bool, NumElements> operator >>(const value_type & lhs, const marray & rhs) = delete;
+
+  friend marray &operator <<=(marray & lhs, const marray & rhs) = delete;
+  friend marray &operator <<=(marray & lhs, const value_type & rhs) = delete;
+
+  friend marray &operator >>=(marray & lhs, const marray & rhs) = delete;
+  friend marray &operator >>=(marray & lhs, const value_type & rhs) = delete;
 
-This proposal adds to the `sycl::ext::oneapi::experimental` namespace, math
-functions accepting the complex types `complex<sycl::half>`, `complex<float>`,
-`complex<double>` as well as the scalar types `sycl::half`, `float` and `double`
-for the SYCL math functions, `abs`, `acos`, `asin`, `atan`, `acosh`, `asinh`,
-`atanh`, `arg`, `conj`, `cos`, `cosh`, `exp`, `log`, `log10`, `norm`, `polar`,
-`pow`, `proj`, `sin`, `sinh`, `sqrt`, `tan`, and `tanh`.
+  friend marray<bool, NumElements> operator <(const marray & lhs, const marray & rhs) = delete;
+  friend marray<bool, NumElements> operator <(const marray & lhs, const value_type & rhs) = delete;
+  friend marray<bool, NumElements> operator <(const value_type & lhs, const marray & rhs) = delete;
+
+  friend marray<bool, NumElements> operator >(const marray & lhs, const marray & rhs) = delete;
+  friend marray<bool, NumElements> operator >(const marray & lhs, const value_type & rhs) = delete;
+  friend marray<bool, NumElements> operator >(const value_type & lhs, const marray & rhs) = delete;
+
+  friend marray<bool, NumElements> operator <=(const marray & lhs, const marray & rhs) = delete;
+  friend marray<bool, NumElements> operator <=(const marray & lhs, const value_type & rhs) = delete;
+  friend marray<bool, NumElements> operator <=(const value_type & lhs, const marray & rhs) = delete;
+
+  friend marray<bool, NumElements> operator >=(const marray & lhs, const marray & rhs) = delete;
+  friend marray<bool, NumElements> operator >=(const marray & lhs, const value_type & rhs) = delete;
+  friend marray<bool, NumElements> operator >=(const value_type & lhs, const marray & rhs) = delete;
+
+  friend marray operator ~(const marray &v) = delete;
+
+  friend marray<bool, NumElements> operator !(const marray &v) = delete;
+};
+
+} // namespace sycl
+```
+
+=== Scalar Mathematical operations
+
+This proposal extends the `sycl::ext::oneapi::experimental` namespace math
+functions to accept `complex<sycl::half>`, `complex<float>`, `complex<double>`
+as well as the scalar types `sycl::half`, `float` and `double` for a range of
+SYCL math functions.
+
+Specifically, it adds support for `abs`, `acos`, `asin`, `atan`, `acosh`,
+`asinh`, `atanh`, `arg`, `conj`, `cos`, `cosh`, `exp`, `log`, `log10`, `norm`,
+`polar`, `pow`, `proj`, `sin`, `sinh`, `sqrt`, `tan`, and `tanh`.
+
+Additionally, this extension introduces support for the `real` and `imag` free
+functions, which the real and imaginary component, respectively.
```

Comment:
```suggestion
functions, which return the real and imaginary component of a number, respectively.
```

I'm not sure, but it may be worth adding a note about what happens if the input is not a complex number here.

Reply:
Don't you think that the declarations of these functions shown after this block (with comments) are not enough?

```c++
/// Returns the real component of the complex number z.                                                                                                                                                                                                                                   
template <class T> constexpr T real(const complex<T> &);                                                                                                                                                                                                                                  
/// Returns the real component of the number y, treated as complex numbers with zero imaginary component.                                                                                                                                                                                 
template <class T> constexpr T real(T);                                                                                                                                                                                                                                                   
/// Returns the imaginary component of the complex number z.                                                                                                                                                                                                                              
template <class T> constexpr T imag(const complex<T> &);                                                                                                                                                                                                                                  
/// Returns the imaginary component of the number y, treated as complex numbers with zero imaginary component.                                                                                                                                                                            
template <class T> constexpr T imag(T); 
```